test_napi/src/lib.rs

```diff
@@ -27,7 +27,7 @@ pub mod typedarray;
 #[macro_export]
 macro_rules! cstr {
   ($s: literal) => {{
-    std::ffi::CString::new($s).unwrap().as_ptr()
+    std::ffi::CString::new($s).unwrap().into_raw()
```

Comment:
the allocate cstring wasn't leaked so the pointer from `.as_ptr` was dangling. `into_raw` leaks it

Reply:
Good catch!